egs/wsj/s5/steps/cleanup/segment_long_utterances.sh

```diff
@@ -0,0 +1,398 @@
+      --input-idf-stats=$dir/docs/idf_stats.txt \
+      --accumulate-over-docs=false \
+      - $dir/docs/split$nj/JOB/query_tf_idf.txt 
+  
+  # The relevant documents can be found using TF-IDF similarity and nearby
+  # documents can also be picked for the Smith-Waterman alignment stage.
+
+  cut -d ' ' -f 1,2 $data_uniform_seg/segments > $data_uniform_seg/utt2reco
+
+  $cmd JOB=1:$nj $dir/lats/log/retrieve_similar_docs.JOB.log \
+    steps/cleanup/internal/retrieve_similar_docs.py \
+      --query-tfidf=$dir/docs/split$nj/JOB/query_tf_idf.txt \
+      --source-tfidf=$dir/docs/split$nj/JOB/src_tf_idf.txt \
+      --source-text-id2doc-ids=$dir/docs/text2doc \
+      --query-id2source-text-id=$data_uniform_seg/utt2reco \
+      --num-neighbors-to-search=$num_neighbors_to_search \
+      --neighbor-tfidf-threshold=$neighbor_tfidf_threshold \
+      --relevant-docs=$dir/docs/split$nj/JOB/relevant_docs.txt
+  
+  $cmd JOB=1:$nj $dir/lats/log/get_ctm_edits.JOB.log \
+    steps/cleanup/internal/stitch_documents.py \
+      --query2docs=$dir/docs/split$nj/JOB/relevant_docs.txt \
+      --input-documents=$dir/docs/split$nj/JOB/docs.txt \
+      --output-documents=- \| \
+    steps/cleanup/internal/align_ctm_ref.py --eps-symbol='"<eps>"' \
+      --oov-word="'`cat $lang/oov.txt`'" --symbol-table=$lang/words.txt \
+      --hyp-format=CTM --align-full-hyp=$align_full_hyp \
+      --reco2file-and-channel=$dir/lats/fake_reco2file_and_channel.JOB \
+      --hyp=$dir/lats/score_$lmwt/${data_id}_uniform_seg.ctm.JOB --ref=- \
+      --output=$dir/lats/score_$lmwt/${data_id}_uniform_seg.ctm_edits.JOB 
+  
+  for n in `seq $nj`; do
+    cat $dir/lats/score_$lmwt/${data_id}_uniform_seg.ctm_edits.$n 
+  done > $dir/lats/score_$lmwt/ctm_edits
+  
+fi
+
+if [ $stage -le 10 ]; then
+  steps/cleanup/internal/resolve_ctm_edits_overlaps.py \
+    ${data_uniform_seg}/segments $dir/lats/score_$lmwt/ctm_edits $dir/ctm_edits
+fi
+
+if [ $stage -le 11 ]; then
+  echo "$0: modifying ctm-edits file to allow repetitions [for dysfluencies] and "
+  echo "   ... to fix reference mismatches involving non-scored words. "
+
+  $cmd $dir/log/modify_ctm_edits.log \
+    steps/cleanup/internal/modify_ctm_edits.py --verbose=3 $dir/non_scored_words.txt \
+    $dir/ctm_edits $dir/ctm_edits.modified
+
+  echo "   ... See $dir/log/modify_ctm_edits.log for details and stats, including"
+  echo " a list of commonly-repeated words."
+fi
+
+if [ $stage -le 12 ]; then
+  echo "$0: applying 'taint' markers to ctm-edits file to mark silences and"
+  echo "  ... non-scored words that are next to errors."
+  $cmd $dir/log/taint_ctm_edits.log \
+       steps/cleanup/internal/taint_ctm_edits.py --remove-deletions=false \
+       $dir/ctm_edits.modified $dir/ctm_edits.tainted
+  echo "... Stats, including global cor/ins/del/sub stats, are in $dir/log/taint_ctm_edits.log."
+fi
+
+if [ $stage -le 13 ]; then
+  echo "$0: creating segmentation from ctm-edits file."
+
+  segmentation_opts=(
```

Comment:
Since almost all of the options of this script are just passed into segment_ctm_edits_mild.py, I wonder whether just having a string-valued --segmentation-opts option would be the easiest; and in comments you could mention two or three of them that you think are the most likely that users might want to override, and point them to that script for further documentation.